examples/interop/RecoveryPlugnWithEndpoints/endpoint.ts

```diff
@@ -0,0 +1,156 @@
+	 */
+	public async addChainForMessageRecovery(chainID: string, wsUrl: string): Promise<void> {
+		await this._configureSidechain(chainID, wsUrl, null);
+
+		const sidechain = (this._sidechainsWithStoreKeys.get(chainID) as SidechainWithStoreKey)
+			.sidechain;
+		await sidechain.saveProofsForMessageRecovery(this._mainchain.getChainID());
+	}
+
+	public async getPendingCCMs(chainID: Buffer): Promise<CCMsg[]> {
+		const eventsDb = new EventsDb(await getDBInstance());
+		return await eventsDb.getCCMsForChainID(chainID);
+	}
+
+	public async triggerMessageRecovery(chainID: string, ccms: CCMsg[]): Promise<string> {
+		const sidechain = this._getSidechain(chainID);
+		if (!sidechain) {
+			throw new Error('No chain was added for message recovery.');
+		}
+
+		try {
+			// return await (new MessageRecovery(this._mainchain, sidechain)).handleFlow(ccms); // returns generated txID
+			return await this._mainchain.handleFlowForMessageRecovery(ccms, sidechain);
+		} catch (err) {
+			return (err as Error).toString();
+		}
+	}
+
+	public async addChainForStateRecovery(
+		chainID: string,
+		wsUrl: string,
+		storeKey: string,
+	): Promise<void> {
+		await this._configureSidechain(chainID, wsUrl, storeKey);
+
+		const sidechain = this._getSidechain(chainID);
+		await sidechain.saveProofsForStateRecovery(Buffer.from(storeKey, 'hex'));
+	}
+
+	public async triggerStateRecovery(chainID: string, storeKey: string): Promise<string> {
+		try {
+			const sidechain = this._getSidechain(chainID);
+			const storeKeys = (this._sidechainsWithStoreKeys.get(chainID) as SidechainWithStoreKey)
+				.storeKeys;
+			if (!storeKeys.includes(storeKey)) {
+				throw new Error('No such storeKey was previously provided for state recovery.');
+			}
+			// return await (new MessageRecovery(this._mainchain, sidechain)).handleFlow(ccms); // returns generated txID
+			return await this._mainchain.handleFlowForStateRecovery(
+				sidechain,
+				Buffer.from(storeKey, 'hex'),
+			);
+		} catch (err) {
+			return (err as Error).toString();
+		}
+	}
+
+	private _getSidechain(chainID: string): Sidechain {
+		const sidechainWithStoreKey = this._sidechainsWithStoreKeys.get(chainID);
+		if (!sidechainWithStoreKey) {
+			throw new Error('No sidechain was configured for recovery.');
+		}
+		return sidechainWithStoreKey.sidechain;
+	}
+
+	// TODO: handle the case when input `chainID` & generated sidechain object's `chainID` don't match with each other.
+	private async _configureSidechain(
+		chainID: string,
+		wsUrl: string,
+		storeKey: string | null,
+	): Promise<void> {
+		if (!this._sidechainsWithStoreKeys.has(chainID)) {
+			const sidechain = new Sidechain();
+			await sidechain.init(this._config, wsUrl, this._messageRecoveryDb, this._stateRecoveryDb);
+
+			if (chainID !== sidechain.getChainID()) {
+				throw new Error("Input chainID doesn't correspond to it's live sidechain.");
+			}
+
+			this._sidechainsWithStoreKeys.set(chainID, { sidechain, storeKeys: [] });
```

Comment:
maybe we should handle the case where if there was an error while connecting to the sidechain , it should not add it to this map